src/Serilog/Core/ILoggingFailureListener.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright © Serilog Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Serilog.Core;
+
+/// <summary>
+/// Implementers can be notified of various failure conditions in the Serilog pipeline.
+/// </summary>
+public interface ILoggingFailureListener
+{
+    /// <summary>
+    /// A logging feature has failed. Often, failures will communicate errors encountered when writing
+    /// log events to sinks.
+    /// </summary>
+    /// <param name="sender">The object that generated or observed the failure.</param>
+    /// <param name="kind">An indication of the failure kind.</param>
+    /// <param name="message">A description of the failure.</param>
+    /// <param name="events">Events associated with the failure.</param>
+    /// <param name="exception">The exception triggering the failure, if any.</param>
+    void OnLoggingFailed(
+        object sender,
+        LoggingFailureKind kind,
+        string message,
+        IReadOnlyCollection<LogEvent>? events,
+        Exception? exception);
```

Comment:
Is there a reason that `Exception` is preferred over `ExceptionDispatchInfo` to preserve the stack trace?

Reply:
Thanks for taking a look!

If the failure is to be forwarded on to another Serilog sink, an `Exception` will be needed; `ExceptionDispatchInfo` is useful for re-throwing, but can't expose enough information to be useful for logging, unfortunately.